NERC-CEH/plankton_ml
# Fastapi model serving

A rough prototype for an API which returns embeddings and classification from a choice of endpoints, each for a different model.

See #52 for notes on the thinking around adding this to one or more pipelines. Definitely an unsophisticated prototype, reusing existing functions plus small parts of https://github.com/alan-turing-institute/ViT-LASNet

## To test

Note this _assumes_ that the model weights for the new Turing models #45 _are not available_ where the test is running!
```
py.test test/test_model_api.py
```

Run the API and send it an image URL, get back embeddings

```
python src/cyto_ml/models/api.py
curl -d url="https://cdn.oceanservice.noaa.gov/oceanserviceprod/facts/nasa-copepod.jpg" http://localhost:8081/resnet50/
```

To run with the Resnet18 weights, see this note https://github.com/alan-turing-institute/ViT-LASNet/tree/main/test and the linked issue here https://github.com/alan-turing-institute/ViT-LASNet/issues/2 - for now a manual stage of downloading them from Google Drive and copying into `./data/weights` - option is to package this as a container
